R/setup.R

```diff
@@ -92,148 +114,196 @@ setup <- function(x_train,
+                           n_MC_samples, seed, keep_samp_for_vS, type, horizon, train_idx, explain_idx,
+                           explain_y_lags, explain_xreg_lags, group_lags = NULL, MSEv_uniform_comb_weights,
+                           verbose, adaptive = FALSE, adaptive_arguments = adaptive_arguments,
+                           shapley_reweighting = "none", testing, is_python, ...) {
+  # Check input type for approach
 
-  if (exact) {
-    internal$parameters$used_n_combinations <- if (is_groupwise) 2^n_groups else 2^n_features
-  } else {
-    internal$parameters$used_n_combinations <-
-      if (is_groupwise) min(2^n_groups, n_combinations) else min(2^n_features, n_combinations)
-    check_n_combinations(internal)
+  # approach is checked more comprehensively later
+  if (!is.logical(paired_shap_sampling) && length(paired_shap_sampling) == 1) {
+    stop("`paired_shap_sampling` must be a single logical.")
   }
 
-  # Check approach
-  check_approach(internal)
+  if (!is.logical(adaptive) && length(adaptive) == 1) {
+    stop("`adaptive` must be a single logical.")
+  }
+  if (!is.list(adaptive_arguments)) {
+    stop("`adaptive_arguments` must be a list.")
+  }
 
-  # Setting default value for n_batches (when NULL)
-  internal <- set_defaults(internal)
 
-  # Checking n_batches vs n_combinations etc
-  check_n_batches(internal)
+  # max_n_coalitions
+  if (!is.null(max_n_coalitions) &&
+      !(is.wholenumber(max_n_coalitions) &&
+        length(max_n_coalitions) == 1 &&
+        !is.na(max_n_coalitions) &&
+        max_n_coalitions > 0)) {
+    stop("`max_n_coalitions` must be NULL or a single positive integer.")
+  }
 
-  # Check regression if we are doing regression
-  if (internal$parameters$regression) internal <- regression.check(internal)
+  # group (checked more thoroughly later)
+  if (!is.null(group) &&
+      !is.list(group)) {
+    stop("`group` must be NULL or a list")
+  }
 
-  return(internal)
-}
+  # n_MC_samples
+  if (!(is.wholenumber(n_MC_samples) &&
+        length(n_MC_samples) == 1 &&
+        !is.na(n_MC_samples) &&
+        n_MC_samples > 0)) {
+    stop("`n_MC_samples` must be a single positive integer.")
+  }
 
-#' @keywords internal
-#' @author Lars Henry Berge Olsen
-regression.check <- function(internal) {
-  # Check that the model outputs one-dimensional predictions
-  if (internal$parameters$output_size != 1) {
-    stop("`regression_separate` and `regression_surrogate` only support models with one-dimensional output")
+  # keep_samp_for_vS
+  if (!(is.logical(keep_samp_for_vS) &&
+        length(keep_samp_for_vS) == 1)) {
+    stop("`keep_samp_for_vS` must be single logical.")
   }
 
-  # Check that we are NOT explaining a forecast model
-  if (internal$parameters$type == "forecast") {
-    stop("`regression_separate` and `regression_surrogate` does not support `forecast`.")
+  # type
+  if (!(type %in% c("normal", "forecast"))) {
```

Comment:
This could be checked in a slightly different way if wanted with 

`type = c("normal", "forecast")` in the arguments and then `type <- match.arg(type)` in the code

Reply:
Thanks, did not know about that function. It seems I cannot use custom error messages, though, so at least I keep it as is for now.